source/features/stop-pjax-loading-with-esc.tsx

```diff
@@ -0,0 +1,91 @@
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+let progressLoader: HTMLElement;
+const progressLoaderLoadingClass = 'is-loading';
+
+function fixProfileNavAndTimeline() {
+	for (const stickyElement of select.all('.js-sticky:not(.is-stuck)')) {
+		stickyElement.removeAttribute('style');
+	}
+}
+
+function keydownHandler(event: KeyboardEvent) {
+	if (event.key !== 'Escape' || !progressLoader.classList.contains(progressLoaderLoadingClass)) {
+		return;
+	}
+
+	if (history.state && '_id' in history.state) {
+		const pjaxContainer = select('#js-repo-pjax-container, #js-pjax-container, #gist-pjax-container');
+
+		if (pjaxContainer) {
+			// We need it for correct work of browser forward-button
+			history.replaceState({
+				url: location.href,
+				title: '',
+				container: `#${pjaxContainer.id}`,
+				...history.state
+			}, '', location.href);
+		} else {
+			features.error(__filebasename, 'Pjax container not found.');
+		}
+	}
+
+	history.back();
+	progressLoader.classList.remove(progressLoaderLoadingClass);
+}
+
+function pjaxErrorHandler(event: CustomEvent) {
+	if (event.cancelable) {
+		// Avoid location.replace() when AbortController.abort() throw an error
+		event.preventDefault();
+	}
+}
+
+function init() {
+	progressLoader = select('.progress-pjax-loader')!;
+
+	window.addEventListener('keydown', keydownHandler);
+
+	window.addEventListener('pjax:error', pjaxErrorHandler);
+
+	if (pageDetect.isUserProfile()) {
+		window.addEventListener('pjax:end', fixProfileNavAndTimeline);
+	}
+}
+
+void features.add({
+	id: __filebasename,
+	description: '',
+	screenshot: false
+}, {
+	include: [
+		pageDetect.isConversationList,
+		pageDetect.isPR,
+		pageDetect.isIssue,
+		pageDetect.isUserProfile,
+		pageDetect.isBlame,
+		pageDetect.isCommit,
+		pageDetect.isRepoCommitList,
+		pageDetect.isCompare,
+		pageDetect.isGlobalSearchResults,
+		pageDetect.isLabelList,
+		pageDetect.isMilestone,
+		pageDetect.isMilestoneList,
+		pageDetect.isNewIssue,
+		pageDetect.isPRConflicts,
+		pageDetect.isQuickPR,
+		pageDetect.isEditingFile,
```

Comment:
I don't think many of these are loaded via ajax. Maybe  `include: isRepo, isGlobalSearchResults` was the right solution after all

Reply:
> I don't think many of these are loaded via ajax.

Yup, BUT these pages contains pjax-links.